### ISSUE TITLE
Invenstigate how RADAR-Rest-Api can be integrated to use authorization with MP 

### ISSUE THREAD

**Issue by nivemaham** [RADAR-CNS/ManagementPortal#25]
We need to investigate the best way to integrate Rest-Api to use security mechanisms of MP to secure REST endpoints in rest-api application.
One cohesive option would be convert Rest-api as a spring project and use it as a `ResourceServer` and use MP's `AuthorizationServer` to ensure the authorization.  (Preferable) 
Otherwise we could provide a small distribution with annotations and integrate it with Rest-api.

**Comment by fnobilia** [comment]
According to this [benchmark](https://github.com/smallnest/Jax-RS-Performance-Comparison) `Jersey` performs better than `Spring`. If possible, I would keep `Rest-Api` in `Jersey`. What about the annotations @dennyverbeeck developed sometime ago?

**Comment by dennyverbeeck** [comment]
Yep, just had a discussion with @nivemaham as well. I'll revisit the implementation to check if we can re-use it to be sure, but i don't expect any issues with it.

**Comment by dennyverbeeck** [comment]
First code added in #36

**Comment by nivemaham** [comment]
@dennyverbeeck Could we manage all the configurations using yaml files, not mix between yaml and env variables?
If i understood it correctly, this should be the distribution we should add as a dependency to rest-api and secure them with Secured annotations?. Did you try the integration with rest-api?

**Comment by dennyverbeeck** [comment]
#39 adds YAML configuration file support to the annotations library. It expects a file `radar_is_config.yml` to be on the classpath and will load from there. Next up will be to finalize the roles implementation, add the roles to the generated JWT, and adapt the annotations implementation based on these roles.

**Comment by dennyverbeeck** [comment]
#59  let clients authenticate with MP, but does not allow for Rest-API to authenticate its own clients. #74 will do the latter, reopening until it is merged.
